Majority Element.java

```diff
@@ -0,0 +1,21 @@
+class Solution 
+{
+    public int majorityElement(int[] nums) 
+    {
+        HashMap<Integer, Integer> map = new HashMap<>();
+        
+        int len = nums.length;
+        
+        for(int n: nums)
+        {
+            int count =  map.getOrDefault(n,0);
+            count++;
+            
+            map.put(n,count);
+            
+            //As soon as the count of number increases by half of numbers in array; we return the number.
+            if(count>(len/2)) return n;
+        }
+        return -1;
+    }
+}
```

Comment:
@HarshitaAgarwal09  You should see "Moore's voting algorithm" will do it in O(1) extra space

Reply:
Okay, will go through it. Thanks.